### ISSUE TITLE
Need some sample maps

### ISSUE THREAD

**Issue by snake-biscuits** [MRVN-Radiant/MRVN-resource-pack#1]
[TF2Maps' A Boojum Snark's Ultimate Mapping Pack](https://tf2maps.net/downloads/ultimate-mapping-resource-pack.510/) contains some great example maps, laying out entities, models & gamemode logic prefabs

It'd be perfect if we could get something like this for [MRVN-Radiant](https://github.com/MRVN-Radiant/MRVN-Radiant)
[TF2Maps.net](https://tf2maps.net) also hosts a number of themed texture packs extending what's available in TF2
Not suggesting we do the same here, but sample maps presenting the textures & models used in each map would be neat

> NOTE: GitHub isn't a file hosting service, we should only upload the essentials

> TODO: break tasks down into sub-issues

Each map should have documentation explaining what files are linked (scripts etc.) & how it works

TASKS:
 - [ ] `TitanfallOnline/maps/`:
    - [ ] `mp_base.map` basic Titanfall Online multiplayer map
       required
       - [ ] spawnpoints for both teams
          - [ ] `info_dropship_start` round start spawns
          - [ ] `info_player_start` pilot spawns w/ classic MP disabled
          - [ ] `info_titan_start` for titan modes & titan resupply
    - [ ] theme zoos showcasing materials, props & even some vingettes (small room or foliage cluster etc.)
       - [ ] `theme_swamp.map` based on `mp_swampland` & ???
 - [ ] `TItanfall2/maps/`:
    - [x] `mp_base.map`  basic Titanfall2 multiplayer map
       required:
       - [x] spawnpoints for both teams
          - [x] `info_dropship_start` round start spawns
          - [x] `info_player_start` pilot spawns w/ classic MP disabled
          - [x] `info_titan_start` for titan modes & titan resupply
    - [ ] `mp_pickups.map` weapon pickups demo (ideally use `.rson` to use built in scripts)
    - [ ] `mp_assembly_line.map` moving platforms & elevators, like Into the Abyss map 2
    - [ ] `mp_gamemode_*.map` prefabs for all the logic needed for each mode
          will also require `.nm` (Recast NavMesh) & `.ain` (Source Engine AI Node Graph) for each
       - [ ] `mp_gamemode_aitdm.map` Attrition
       - [ ] `mp_gamemode_fd.map` Frontier Defense
       - [ ] `mp_gamemode_ffa.map` Free For All
       - [ ] `mp_gamemode_lf.map` Live Fire
    - [ ] `sp_base.map` base essentials for a single-player map
      - [ ] spawnpoint
      - [ ] at least one checkpoint
      - [ ] `trigger_level_transition` to end the level
 - [ ] `ApexLegends/maps/`
   - [ ] `mp_rr_base.map` basic BR map
   - [ ] `mp_rr_train.map` World's Edge train example & breakdown

**Comment by snake-biscuits** [comment]
[MRVN-entities](https://github.com/MRVN-Radiant/MRVN-entities) has now reached the point where we can experiment with entity definitions & start exploring how each entity functions

We should be able to build small maps / sections of an entity zoo to explore how each entity functions

VIS falls apart when map geo gets complex / large, but load of entities should be fine so long as we respect edict limits

**Comment by snake-biscuits** [comment]
r1o theme zoos could prove difficult, as currently textures do not get loaded / cached for some unknown reason
r2 zoos might require figuring out how to cache a map's specific rpak, idk how that works
maybe Northstar mods can tell the filesystem to cache internal rpaks? iirc Icepick mounts everything

Filesystem is pain and could probably be documented better, we best talk to Northstar and TFO:Revive people about fs docs

**Comment by snake-biscuits** [comment]
`maps/r2/mp_base.map` should be really easy
be sure to write up a guide to go along with it

**Comment by snake-biscuits** [comment]
For an example of scope per-example map, here's all 61 `sdk_` sample map names from the Source SDK:
```
$ tree -P "sdk_*.vmf" --prune
.
├── cstrike
│   └── mapsrc
│       ├── sdk_3d_skybox.vmf
│       ├── sdk_de_cbble.vmf
│       ├── sdk_lightmaps.vmf
│       └── sdk_water.vmf
├── dod
│   └── mapsrc
│       ├── sdk_dod_flash.vmf
│       └── sdk_example_hdr.vmf
├── hl2
│   └── mapsrc
│       ├── sdk_3d_skybox.vmf
│       ├── sdk_background.vmf
│       ├── sdk_cables.vmf
│       ├── sdk_cubemaps.vmf
│       ├── sdk_d1_town_03.vmf
│       ├── sdk_d1_trainstation_01.vmf
│       ├── sdk_d1_trainstation_02.vmf
│       ├── sdk_d1_trainstation_03.vmf
│       ├── sdk_d1_trainstation_04.vmf
│       ├── sdk_d1_trainstation_05.vmf
│       ├── sdk_d2_coast_12.vmf
│       ├── sdk_d3_c17_13.vmf
│       ├── sdk_d3_citadel_01.vmf
│       ├── sdk_faceposer.vmf
│       ├── sdk_func_detail.vmf
│       ├── sdk_hints.vmf
│       ├── sdk_lightmaps.vmf
│       ├── sdk_materials.vmf
│       ├── sdk_occluders.vmf
│       ├── sdk_phys_ballsocket.vmf
│       ├── sdk_phys_constraint.vmf
│       ├── sdk_phys_convert.vmf
│       ├── sdk_phys_explosion.vmf
│       ├── sdk_phys_hinge.vmf
│       ├── sdk_phys_impact.vmf
│       ├── sdk_phys_keepupright.vmf
│       ├── sdk_phys_lengthconstraint.vmf
│       ├── sdk_phys_magnet.vmf
│       ├── sdk_phys_motor.vmf
│       ├── sdk_phys_pulleyconstraint.vmf
│       ├── sdk_phys_ragdollconstraint.vmf
│       ├── sdk_phys_ragdollmagnet.vmf
│       ├── sdk_phys_slideconstraint.vmf
│       ├── sdk_phys_spring.vmf
│       ├── sdk_phys_thruster.vmf
│       ├── sdk_phys_torque.vmf
│       ├── sdk_prop_types.vmf
│       ├── sdk_shader_samples.vmf
│       ├── sdk_vehicles.vmf
│       └── sdk_water.vmf
├── hl2mp
│   └── mapsrc
│       └── sdk_dm_lockdown.vmf
├── sdk
│   └── maps
│       ├── sdk_teams_hdr.vmf
│       └── sdk_vehicles.vmf
└── tf
    └── mapsrc
        ├── sdk_arena_lumberyard.vmf
        ├── sdk_arena_ravine.vmf
        ├── sdk_cp_badlands.vmf
        ├── sdk_cp_dustbowl.vmf
        ├── sdk_cp_granary.vmf
        ├── sdk_cp_gravelpit.vmf
        ├── sdk_ctf_2fort.vmf
        ├── sdk_pl_badwater.vmf
        ├── sdk_pl_goldrush.vmf
        ├── sdk_tc_hydro.vmf
        ├── sdk_tr_dustbowl.vmf
        └── sdk_tr_target.vmf

12 directories, 61 files
```
`zoo_ziplines.map` should make a decent equivalent
sdk versions of official maps (decompiles) shouldn't be distributed imo, but select samples from official maps would be very useful
more effort, less distributing copyrighted material without permission

**Comment by F1F7Y** [comment]
While we dont want to ship the entirety of mp_common I think we could ship `world/dev/dev_gray_512` to make these maps look better.

Obviously we'll want to ship our own base assets with the modding clients down the line.

**Comment by snake-biscuits** [comment]
> we could ship `world/dev/dev_gray_512` to make these maps look better.

Good idea imo, using rpak textures so we can play with lighting would be great in general

I might have it set up in MRVN already, I’ll have to check

But as was mentioned in discord, I’m having a skill issue with textures atm
